src/electionguard_gui/components/key_ceremony_details_component.py

```diff
@@ -58,30 +70,78 @@ def watch_key_ceremony(self, key_ceremony_id: str) -> None:
         )
 
     def on_key_ceremony_changed(self, key_ceremony_id: str) -> None:
-        is_admin = self.auth_service.is_admin()
+        current_user_id = self._auth_service.get_user_id()
+        self.log.debug(
+            f"on_key_ceremony_changed key_ceremony_id: '{key_ceremony_id}', current_user_id: '{current_user_id}'"
+        )
+        is_admin = self._auth_service.is_admin()
+        is_guardian = not is_admin
         db = self.db_service.get_db()
         key_ceremony = self.get_ceremony(db, key_ceremony_id)
-        guardians_joined_count = len(key_ceremony["guardians_joined"])
-        guardian_count = key_ceremony["guardian_count"]
-        self.log.info(
-            f"{guardians_joined_count}/{guardian_count} guardians joined {key_ceremony_id}"
-        )
-        all_guardians_joined = guardians_joined_count >= guardian_count
-        other_keys_exist = len(key_ceremony["other_keys"]) > 0
-        if is_admin and all_guardians_joined and not other_keys_exist:
+        state = self._ceremony_state_service.get_key_ceremony_state(key_ceremony)
```

Comment:
I hope to eventually refactor much of this code into a finite state machine pattern.  For now, I've at least refactored the logic that determines which state you're in into this key ceremony state service.